src/sdk/tx/account.ts

```diff
@@ -0,0 +1,40 @@
+import { decodeOptionalPubkey } from "@cosmjs/proto-signing"
+import { Account, accountFromAny, AccountParser } from "@cosmjs/stargate"
+import { EthAccount } from "src/protojs/eth/types/v1/account"
+import { Any } from "src/protojs/google/protobuf/any"
+import { assert } from "@cosmjs/utils"
+import { BaseAccount } from "src/protojs/cosmos/auth/v1beta1/auth"
+
+/**
+ * Converts an EthAccount to a general Cosmos Account object.
+ *
+ * @param {EthAccount} ethAccount - The EthAccount object containing the account's base information.
+ * @returns {Account} The Cosmos account object.
+ */
+export const accountFromEthAccount = (baseAccount: BaseAccount): Account => {
+  const { address, pubKey, accountNumber, sequence } = baseAccount
```

Comment:
Could destructure in the function instead

const fn = ({address, pub,...}: BaseAccount) => {...}

Reply:
changed